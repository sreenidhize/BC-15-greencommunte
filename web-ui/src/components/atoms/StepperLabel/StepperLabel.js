import { makeStyles, Typography } from "@material-ui/core";
import React from "react";

const useStyles = makeStyles((theme) => ({
  root: {
    fontFamily: theme.typography.fontFamily,
    fontWeight: 500,
  },
}));

const StepperLabel = (props) => {
  const classes = useStyles();
  return (
    <Typography
      style={{
        color: props.isVisited ? "#5ac568" : "#5f7381",
        fontSize: "18px",
        marginRight: "15px",
      }}
      className={classes.root}
      data-testid="stepper-label"
    >
      {props.label}
    </Typography>
  );
};

export default StepperLabel;
