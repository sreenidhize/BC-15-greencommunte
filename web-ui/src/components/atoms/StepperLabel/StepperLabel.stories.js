import { ThemeProvider } from "@material-ui/core";
import React from "react";
import Label from "./StepperLabel";
import theme from "../../../utils/theme/theme";

export default {
  title: "atoms/StepperLabel",
  component: Label,
};

const Template = (args) => (
  <ThemeProvider theme={theme}>
    <Label {...args} />
  </ThemeProvider>
);

export const Primary = Template.bind({});

Primary.args = {
  label: "Your Location",
  isVisited: false,
};
