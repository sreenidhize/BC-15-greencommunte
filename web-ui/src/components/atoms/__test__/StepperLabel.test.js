import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import StepperLabel from "../StepperLabel/StepperLabel";

describe("Stepper Label test", () => {
    it("Matches the snapshot", () => {  
      const { container } = render(
        <StepperLabel isVisited={true} label="Your Skills" />
      );
      expect(container).toMatchSnapshot();
    });
    test("Test Stepper Label render", () => {
      const { getByTestId } = render(<StepperLabel isVisited={false} label="Your Location"  />);
      expect(getByTestId("stepper-label")).toBeInTheDocument();
    });
  
    test("Test Stepper visited Label render", () => {
      const { getByTestId } = render(<StepperLabel isVisited={true} label="Job Location" />);
      expect(getByTestId("stepper-label")).toBeInTheDocument();
    });
  });