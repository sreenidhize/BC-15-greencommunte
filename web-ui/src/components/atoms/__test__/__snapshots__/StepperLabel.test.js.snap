// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper Label test Matches the snapshot 1`] = `
<div>
  <p
    class="MuiTypography-root makeStyles-root-1 MuiTypography-body1"
    data-testid="stepper-label"
    style="color: rgb(90, 197, 104); font-size: 18px; margin-right: 15px;"
  >
    Your Skills
  </p>
</div>
`;
